cmd/podman/down/kube.go

```diff
@@ -0,0 +1,94 @@
+package down
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/containers/podman/v4/cmd/podman/common"
+	"github.com/containers/podman/v4/cmd/podman/registry"
+	"github.com/containers/podman/v4/cmd/podman/utils"
+	"github.com/containers/podman/v4/pkg/domain/entities"
+	"github.com/spf13/cobra"
+)
+
+// playKubeOptionsWrapper allows for separating CLI-only fields from API-only
+// fields.
+type playKubeOptionsWrapper struct {
+	entities.PlayKubeDownOptions
+}
+
+var (
+	kubeOptions     = playKubeOptionsWrapper{}
+	kubeDescription = `Command stops pods or volumes defined in Kubernetes YAML.
+
+  It stops pods or volumes created from a Kubernetes YAML. Supported kinds are Pods, Deployments and PersistentVolumeClaims.`
+
+	kubeCmd = &cobra.Command{
+		Use:               "kube [options] KUBEFILE|-",
+		Short:             "Stop a pod or volume based on Kubernetes YAML.",
+		Long:              kubeDescription,
+		RunE:              kube,
+		Args:              cobra.ExactArgs(1),
+		ValidArgsFunction: common.AutocompleteDefaultOneArg,
+		Example: `podman down kube nginx.yml
+  cat nginx.yml | podman down kube -
+  podman down kube --creds user:password --seccomp-profile-root /custom/path apache.yml`,
+	}
+)
+
+func init() {
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Command: kubeCmd,
+		Parent:  downCmd,
+	})
+
+	flags := kubeCmd.Flags()
+	flags.SetNormalizeFunc(utils.AliasFlags)
+}
+
+func kube(cmd *cobra.Command, args []string) error {
+	yamlfile := args[0]
+	if yamlfile == "-" {
+		yamlfile = "/dev/stdin"
+	}
+
+	return teardown(yamlfile)
+}
+
+func teardown(yamlfile string) error {
```

Comment:
Unfortunately, this is copied/pasted from `play`. Any suggestions for how to reduce this duplication? I could potentially make these functions public, and invoke them from `down`; or define the `down` commands in the same package as `play`.

Reply:
put it into common ?

Reply:
A thought that occurred to me while I was moving things around: it seems we might want to pass the cobra.Command `cmd` as the first parameter, then write to `cmd.OutOrStdout()` an duse `cmd.Context()`? I'm not sure how this code works, so am unsure what the difference is with `registry.GetContext()`

Reply:
You can just write to stdout, cmd.OutOrStdout() is not used anywhere AFAIK.